traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -426,10 +439,20 @@ func (cg *TOCacheGroup) Read() ([]interface{}, error, error, int) {
 		}
 		s.LocalizationMethods = &lms
 		s.Fallbacks = &cgfs
+		// In case of a bulk read, even if one of the items has a "lastUpdated" time that is after whats supplied in the request,
+		// we send back the entire array of results
+		if !s.LastUpdated.Before(modifiedSince) {
```

Comment:
This isn't sufficient. If objects have been deleted, the endpoint won't be flagged as Modified, and will return a 304 when there have actually been changes.

This needs to keep track of deleted rows somehow.

There are several ways to do it. One would be to create a `deleted_foo` table for each existing table, tracking deletions. Possibly with a cleanup routine, and logic to determine "if IMS request  is older than the oldest deletion/cleanup time, consider it modified.

Another way is to make deletions update a "last modified" timestamp  on the endpoint somehow. This could be a simple table of timestamps for each table, or it could artificially update the timestamp of some row on another table which is part of the endpoint (for example, deliveryservices_regexes could update the timestamp of the deliveryservice they belong to, when one  is deleted). Possibly as a Database Trigger.

Those are just ideas. As long as deletions are tracked, I don't have a strong opinion on how.

Reply:
Great catch, I'll work on fixing that